AP_HAL_ChibiOS: add AMOV Flycore board support (#33419)
ArduPilot/ardupilot: libraries/AP_HAL_ChibiOS/hwdef/flycore/README.md

interfaces shown in the pinout image.

Pin 1 starts at the right side of each connector when viewed as shown below.

![AMOV Flycore pin 1 orientation](amovlab_flycore_pin1_orientation.png "Connector pin 1 orientation")

### CAN1

| Pin | Signal |
| --- | ------ |
| 1 | CAN_L |
| 2 | CAN_H |
| 3 | GND |
| 4 | VCC |

### GPS2

| Pin | Signal |
| --- | ------ |
| 1 | RX |
| 2 | TX |
| 3 | GND |
| 4 | VCC |

### TELEM3

| Pin | Signal |
| --- | ------ |
| 1 | RX |
| 2 | TX |
| 3 | GND |
| 4 | VCC |

### I2C1 and I2C4

| Pin | Signal |
| --- | ------ |
| 1 | SDA |
| 2 | SCL |
| 3 | GND |
| 4 | VCC |

### TELEM2

| Pin | Signal |
| --- | ------ |
| 1 | GND |
| 2 | RX |
| 3 | TX |

### PWM

| Pin | Signal |
| --- | ------ |
| 1 | GND |
| 2 | BUZZER |
| 3 | FMU_CH8 |
| 4 | FMU_CH7 |
| 5 | FMU_CH6 |
| 6 | FMU_CH5 |
| 7 | FMU_CH4 |
| 8 | FMU_CH3 |
| 9 | FMU_CH2 |
| 10 | FMU_CH1 |

## UART Mapping

| Serial | UART | Port | Default protocol |
| ------ | ---- | ---- | ---------------- |
| SERIAL0 | OTG1 | USB | MAVLink2 |
| SERIAL1 | USART2 | TELEM1 | MAVLink2 |
| SERIAL2 | USART3 | TELEM2 | MAVLink2 |
| SERIAL3 | UART4 | GPS1, onboard UM982 | GPS |

> | SERIAL4 | UART5 | TELEM3 | MAVLink2 |

Reviewer: set to MAVLink2 in hwdef, currently GPS protocol

> | SERIAL5 | USART6 | RC | RC input |

Reviewer: need to sset to RCIN protocol, currently NONE

> | SERIAL6 | UART7 | GPS2 | GPS |

Reviewer: you need to set this as GPS protocol in the hwdef...its currently NONE

> | SERIAL7 | UART8 | Debug pads | Disabled |

Reviewer: UART 8 is definitely defined...since its not pinned out, the comment should be "not pinned out"...user will see SERAIL8 in params

GPS1 is connected internally to the onboard UM982 and is not an external
connector. GPS2 is the external GPS connector.

## RC Input

> The RC connector is connected to USART6 and supports serial RC protocols

Reviewer: there is not RC connector shown in the pinout....

> such as SBUS, DSM/DSMX, CRSF, and GHST. PPM input is available on PC8.

Reviewer: there is no PC8 in the pinout image...and since its a RC interrupt input, it supports all unidirectional serial protocols including PM...UART6(wherever that is pinned out) would support the bi-directional protocols you list

> Bi-directional protocols require both the RX and TX signals.

Reviewer: ```suggestion
Bi-directional protocols require both the RX and TX signals. See [RC Systems](https://ardupilot.org/copter/docs/common-rc-systems.html) for more details on setup.
```

## PWM Outputs

The Flycore supports 10 FMU PWM outputs in three timer groups:

- Outputs 1 to 4 use TIM1.
- Outputs 5 to 7 use TIM3.
- Outputs 8 to 10 use TIM4.

Outputs in the same timer group must use the same output rate and protocol.
Outputs 1 to 8 are on the PWM connector. Outputs 9 and 10 are available on
the separate pads shown in the pinout image.
The PWM rail is externally powered; the flight controller only supplies the
PWM signals.

## GPIOs

| Function | GPIO number |
| -------- | ----------- |

> | FMU_CH1 to FMU_CH10 | 50 to 59 |

Reviewer: Function column should be pin names and match pinout

| Buzzer | 77 |

> | IMU heater | 80 |

Reviewer: all the followingshould be removed....not useful to a user...I dont know what SBUS output enable does...

| Hardware version drive | 82 |
| SBUS output enable | 83 |
| Red LED | 90 |
| Blue LED | 92 |

> ## Analog Inputs

Reviewer: This section should only reflect the user interfacing pins...and I dont even see an RSSI pin in the pnout

The board routes RSSI, hardware identification, 5 V sensing, battery voltage,
and battery current signals to ADC pins. ArduPilot currently enables RSSI,
hardware identification, and 5 V sensing. Battery monitoring is not enabled
until the voltage divider and current sensor calibration are verified.

## Compass

The Flycore has no onboard compass. An external compass can be connected to

> the I2C1 or I2C4 connector. The GPS1 and GPS2 serial interfaces do not include

Reviewer: ```suggestion
the I2C1 or I2C4 connector. 
```

> compass I2C signals.

Reviewer: remove

## Firmware

Firmware for the Flycore is available from the
[ArduPilot firmware server](https://firmware.ardupilot.org/) under the
`flycore` target.

> To build the firmware from source:

Reviewer: remove build instructions...uneeded

```sh
./waf configure --board flycore
./waf copter
```

> ## Loading Firmware

Reviewer: does this come with AP pre-loaded? if so these instructions are should simply be to use a GCS to update using the apj file...if not need DFU instructions (and where the boot button is located)

# AMOV Flycore Flight Controller

The AMOV Flycore is an STM32H743-based flight controller manufactured by
[AMOV](https://amovlab.com/). It includes an onboard UM982 GNSS receiver,
dual IMUs, an onboard barometer, 10 PWM outputs, dual CAN, three telemetry
ports, and a standard SD card slot.

![AMOV Flycore](amovlab_flycore_product.jpg "AMOV Flycore")

## Where to Buy

For purchasing information, contact
[shudajun@amovauto.com](mailto:shudajun@amovauto.com).

## Specifications

- Processor
  - STM32H743, Arm Cortex-M7 at 480 MHz
  - 2 MB flash and 1 MB RAM
  - No I/O coprocessor
- Sensors
  - Bosch BMI088 IMU
  - InvenSense ICM-42688P IMU
  - MS5611 barometer
  - UM982 GNSS receiver connected internally to GPS1
  - No onboard compass
- Interfaces

>   - 10 FMU PWM outputs

Reviewer: ```suggestion
  - 8 PWM outputs
```

Author: Updated the board overview to list eight PWM outputs.

The external peripheral signal connectors use JST-GH connectors with
1.25 mm pitch unless otherwise noted. The POWER connector is
XT30PW(2+2)-M. USB, the SD card slot, RTK antenna connectors, and the
exposed buzzer, SWD, and debug pads use the interfaces shown in the
pinout image.

Pin 1 starts at the right side of each connector when viewed as shown below.

![AMOV Flycore pin 1 orientation](amovlab_flycore_pin1_orientation.png "Connector pin 1 orientation")

The VCC pins on signal connectors provide 5 V unless noted otherwise.

### CAN1

| Pin | Signal |
| --- | ------ |
| 1 | CAN_L |
| 2 | CAN_H |
| 3 | GND |
| 4 | VCC |

### I2C1

| Pin | Signal |
| --- | ------ |
| 1 | SDA |
| 2 | SCL |
| 3 | GND |
| 4 | VCC |

### I2C4

| Pin | Signal |
| --- | ------ |
| 1 | SDA |
| 2 | SCL |
| 3 | GND |
| 4 | VCC |

### TELEM2

| Pin | Signal |
| --- | ------ |
| 1 | GND |
| 2 | RX |
| 3 | TX |

### RTK COM2

| Pin | Signal |
| --- | ------ |
| 1 | RX |
| 2 | TX |
| 3 | GND |
| 4 | PPS |

### TELEM1

| Pin | Signal |
| --- | ------ |
| 1 | RX |
| 2 | TX |
| 3 | GND |
| 4 | VCC |

### SBUS

| Pin | Signal |
| --- | ------ |
| 1 | RX |
| 2 | TX |
| 3 | GND |
| 4 | VCC |

> The SBUS connector is connected to USART6. It supports serial RC protocols

Reviewer: ```suggestion
## RC Input

The SBUS connector is connected to USART6. It supports all RC protocols except PPM.
```

Author: Added an RC Input section documenting that the SBUS connector is connected to USART6 and that PPM is not exposed.

> such as SBUS, DSM/DSMX, CRSF, and GHST. Bi-directional protocols require

Reviewer: ```suggestion
Bi-directional protocols require
```

Author: Updated the description to state that bi-directional protocols require both RX and TX, and retained the setup reference.

both the RX and TX signals. See
[RC Systems](https://ardupilot.org/copter/docs/common-rc-systems.html) for
more details on setup. PPM is not brought out to an external connector.

### GPS2

| Pin | Signal |
| --- | ------ |
| 1 | RX |
| 2 | TX |
| 3 | GND |
| 4 | VCC |

### TELEM3

| Pin | Signal |
| --- | ------ |
| 1 | RX |
| 2 | TX |
| 3 | GND |
| 4 | VCC |

### PWM

| Pin | Signal |
| --- | ------ |
| 1 | GND |
| 2 | BUZZER |

> | 3 | FMU_CH8 |

Reviewer: all of  these should just be CHx to match pinout

Author: Renamed the PWM connector signals to CH1 through CH8 to match the pinout.

| 4 | FMU_CH7 |
| 5 | FMU_CH6 |
| 6 | FMU_CH5 |
| 7 | FMU_CH4 |
| 8 | FMU_CH3 |
| 9 | FMU_CH2 |
| 10 | FMU_CH1 |

The board also has exposed pads labelled M0-P/M0-N through M3-P/M3-N.
These pads are not used by ArduPilot on the Flycore.

### POWER

> 

Reviewer: ```suggestion
The power input is setup for a DroneCan battery monitor.
```
you should add a defaults.parm enabling whichever CAN bus this is on and add the batt mon def in the hwdef for a DroneCan monitor ("8")

Author: Added defaults.parm with CAN_P1_DRIVER 1 and BATT_MONITOR 8. The README now documents the CAN1 DroneCAN battery-monitor connection.

| Pin | Signal |
| --- | ------ |
| 1 | GND |
| 2 | VCC |
| 3 | CAN_L |
| 4 | CAN_H |

The POWER connector is XT30PW(2+2)-M. VCC is the main power input and
supports 15 V to 28 V.

### FAN Power

| Pin | Signal |
| --- | ------ |
| 1 | GND |
| 2 | VCC |

The fan power connector provides 5 V.

### Debug and SWD pads

| Pad | Signal |
| --- | ------ |
| 3V3 | 3.3 V |
| GND | GND |
| SWD | SWDIO |
| SWC | SWCLK |
| DRX | UART8 RX |
| DTX | UART8 TX |

SWD and SWC are used for bootloader flashing and debugging. DRX and DTX
are the UART8 debug serial pins.

## UART Mapping

| Serial | UART | Port | Default protocol |
| ------ | ---- | ---- | ---------------- |
| SERIAL0 | OTG1 | USB | MAVLink2 |
| SERIAL1 | USART2 | TELEM1 | MAVLink2 |
| SERIAL2 | USART3 | TELEM2 | MAVLink2 |
| SERIAL3 | UART4 | GPS1, onboard UM982 | GPS |
| SERIAL4 | UART5 | TELEM3 | MAVLink2 |
| SERIAL5 | USART6 | SBUS | RC input |
| SERIAL6 | UART7 | GPS2 | GPS |
| SERIAL7 | UART8 | Debug pads | Disabled |
| SERIAL8 | OTG2 | Not pinned out | Disabled |

GPS1 is connected internally to the onboard UM982 and is not an external
connector. GPS2 is the external GPS connector.

## PWM Outputs

> The Flycore supports 10 FMU PWM outputs in three timer groups:

Reviewer: ```suggestion
The Flycore supports 8 FMU PWM/Dshot outputs on the PWM connector, in three groups:
```

Author: Updated as suggested. The README now describes the eight exposed FMU PWM/DShot outputs on the PWM connector in three groups.

> - Outputs 1 to 4 use TIM1.

Reviewer: ```suggestion
- CH1-CH4 Group1
```

Author: Updated the Group 1 label to match the pinout.

> - Outputs 5 to 7 use TIM3.

Reviewer: ```suggestion
- CH5- CH7 Group2
```

Author: Updated the Group 2 label to match the pinout.

> - Outputs 8 to 10 use TIM4.

Reviewer: ```suggestion
- CH8 Group3
```

Author: Updated the Group 3 label to match the pinout.

Outputs in the same timer group must use the same output rate and protocol.

> Outputs 1 to 8 are on the PWM connector. Outputs 9 and 10 are configured in

Reviewer: remove line

Author: Removed this line.

> the hardware definition but are not exposed on the main PWM connector.

Reviewer: Remove line

Author: Removed this line.
